src/EditorFeatures/Core/Extensibility/VersionSelector.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Extensibility.Composition;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    internal sealed class VersionSelector
+    {
+        public static T Select<T>(IEnumerable<Lazy<T, VisualStudioVersionMetadata>> items)
+        {
+            var best = items.FirstOrDefault();
```

Comment:
`items.OrderByDescending(i => i.Metadata.Version).FirstOrDefault()`? Given items probably won't be more than few entries, this might be easier.